Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -658,3 +718,304 @@ class ChannelUpdater: Worker {
+            startWatching(cid: cid, isInRecoveryMode: isInRecoveryMode) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func stopWatching(cid: ChannelId) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            stopWatching(cid: cid) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func truncateChannel(cid: ChannelId, skipPush: Bool, hardDelete: Bool, systemMessage: String?) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            truncateChannel(cid: cid, skipPush: skipPush, hardDelete: hardDelete, systemMessage: systemMessage) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+
+    @discardableResult func update(
+        channelQuery: ChannelQuery,
+        memberSorting: [Sorting<ChannelMemberListSortingKey>] = []
+    ) async throws -> ChannelPayload {
+        // Just populate the closure since we select the endpoint based on it.
+        let useCreateEndpoint: ((ChannelId) -> Void)? = channelQuery.cid == nil ? { _ in } : nil
+        let actions = ChannelUpdateActions(
+            memberListSorting: memberSorting,
+            resetMembers: true,
+            resetWatchers: true
+        )
+        return try await withCheckedThrowingContinuation { continuation in
+            update(
+                channelQuery: channelQuery,
+                isInRecoveryMode: false,
+                onChannelCreated: useCreateEndpoint,
+                actions: actions,
+                completion: continuation.resume(with:)
+            )
+        }
+    }
+    
+    func update(channelPayload: ChannelEditDetailPayload) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            updateChannel(channelPayload: channelPayload) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func updatePartial(channelPayload: ChannelEditDetailPayload, unsetProperties: [String]) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            partialChannelUpdate(updates: channelPayload, unsetProperties: unsetProperties) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func uploadFile(type: AttachmentType, localFileURL: URL, cid: ChannelId, progress: ((Double) -> Void)? = nil) async throws -> UploadedAttachment {
+        try await withCheckedThrowingContinuation { continuation in
+            uploadFile(type: type, localFileURL: localFileURL, cid: cid, progress: progress) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    // MARK: -
+    
+    func loadMessages(with channelQuery: ChannelQuery, pagination: MessagesPagination) async throws -> [ChatMessage] {
```

Comment:
can you please remind me why we needed these 3 methods below?

Reply:
No other reason than keeping the `Chat.swift` has clean as possible